pkg/proxy/endpointslicecache.go

```diff
@@ -113,15 +112,11 @@ func newEndpointSliceTracker() *endpointSliceTracker {
 // newEndpointSliceData generates endpointSliceData from an EndpointSlice.
 func newEndpointSliceData(endpointSlice *discovery.EndpointSlice, remove bool) *endpointSliceData {
 	esData := &endpointSliceData{
-		Ports:     make([]discovery.EndpointPort, len(endpointSlice.Ports)),
+		Ports:     endpointSlice.Ports,
 		Endpoints: []*endpointData{},
 		Remove:    remove,
 	}
 
-	// copy here to avoid mutating shared EndpointSlice object.
-	copy(esData.Ports, endpointSlice.Ports)
-	sort.Sort(byPort(esData.Ports))
```

Comment:
> the EndpointSlice controller and mirroring controller always output the ports in the same order

there can be other controllers creating EndpointSlices or custom ones,

Reply:
if someone adds labels or annotations to the slice, or similar changes ?  is this not premature optimization? I never seen any issue about proxy  memory usage ...

Reply:
It still [optimizes out updates that don't result in a different BaseEndpointInfo](https://github.com/kubernetes/kubernetes/blob/05e14799db8e4835d686d5b604ea4c9f5708120d/pkg/proxy/endpointslicecache.go#L266), so label/annotation changes will still have no effect.

> is this not premature optimization?

The _existing_ code is the premature optimization: it is doing extra work to optimize a case that isn't expected to happen. This PR removes that premature optimization and makes the code simpler.

Reply:
ok , fair, let me review in detail, I recognize I'm being lazy, sorry :)